### PR TITLE
660 spatial audio implementation for web

### LINKED ISSUE #660: Spatial audio implementation for web
If you look here: https://console.agora.io/marketplace/extension/introduce?serviceName=%20agora-spatial-audio
There is a web download and demo code

### LINKED ISSUE #761: Need to add the "Web SDK" details for 3D Spatial Audio in New Documentation Center under Spatial Audio
Ref of the file location in New doc center : https://docs.agora.io/en/voice-calling/develop/spatial-audio

### PR DESCRIPTION
**Pull request by saudsami**
closes #660 
closes #761

### CODE REVIEW

File: shared/video-sdk/develop/spatial-audio/project-implementation/web.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<PlatformWrapper platform="web">
+
+### Spatial audio for remote users
+
+When subscribing to a remote user, realize the user's spatial sound effect as follows:
+
+1. Call [`createProcessor`](#createprocessor) to create a `SpatialAudioProcessor` instance.
```
Comment (billy-the-fish) [inline shared/video-sdk/develop/spatial-audio/project-implementation/web.mdx:7]: So? 

Where is the code example, this does not fit the pattern. 

. Some kind of explanation
   code example. 


I'm stopping here. Can you have a quick look at the Android guide and update to look like that. Soon as that is done I will review again.

**Review by billy-the-fish — CHANGES_REQUESTED** [review]
I've left a few comments. As a global idea, please have another look a the Android guide and update to match.

**Comment by billy-the-fish** [comment]
Can you have a look at the code in https://github.com/AgoraIO/Docs-Source/issues/894 as an idea on how to update this so the spatial audio feature is easier to detect.

File: shared/video-sdk/develop/_spatial-audio.mdx [hunk 2]
```diff
@@ -31,20 +33,42 @@ You can configure the following spatial audio effects:
...
+
+![Spatial Audio](/images/video-sdk/spatial-audio-web-mediaplayer.svg)
+</PlatformWrapper>
 
 ## Prerequisites
 
 In order to follow this procedure you must have:
 
 * Implemented <Link to="../get-started/get-started-sdk">Get Started with <Vpd k="NAME"/></Link>
+<PlatformWrapper platform='web'>
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/_spatial-audio.mdx:62]: done

File: shared/video-sdk/develop/spatial-audio/project-setup/web.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    To enable spatial audio functionality, add the Agora spatial audio extension to your project.
+    
+    1.	Run the following command to install the spatial audio plugin:
+
+        ```bash
+        npm install agora-extension-spatial-audio
+        ```
+
+    1.	Add the following code to your  `.js` file to import the spatial audio module:
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/spatial-audio/project-setup/web.mdx:13]: done.

File: shared/video-sdk/develop/spatial-audio/project-setup/web.mdx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<PlatformWrapper platform="web">
+
+1.	**Integrate the spatial audio plugin** 
+
+    To enable spatial audio functionality, add the Agora spatial audio extension to your project.
+    
+    1.	Run the following command to install the spatial audio plugin:
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/spatial-audio/project-setup/web.mdx:7]: done

File: shared/video-sdk/develop/spatial-audio/project-setup/web.mdx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        ```javascript
+        import {SpatialAudioExtension} from agora-extension-spatial-audio;
+        ```
+
+1.	**Dynamically load Wasm and JS file dependencies** 
+    
+    The Spatial audio plugin depends on some Wasm and JS files. To ensure that the browser can load and run these files properly, take the following steps:
+    
+    1.	Publish the Wasm and JS files in the  `node_modules/agora-extension-spatial-audio/external` directory to a CDN or static resource server, and ensure that they are in the same public path. When creating an  `SpatialAudioExtension` object, you pass in the URL of the public path, and the plugin dynamically loads the dependent files.
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/spatial-audio/project-setup/web.mdx:23]: done

File: shared/video-sdk/develop/spatial-audio/project-setup/web.mdx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+        * If the Host URL of the Wasm and JS files is inconsistent with the Host URL of the web application, you need to enable the CORS policy for accessing the file domain name.
+
+        * You cannot put Wasm and JS files under HTTP service because loading HTTP resources under HTTPS domain name is prohibited by browser security policy.
+
+    1.	If Content Security Policy (CSP) is enabled, Chrome and Edge prohibit loading and running Wasm files by default. You need to configure CSP according to the browser version:
+
+        * Versions greater than or equal to Chrome 97 and Edge 97: Add `script-src` in options `wasm-unsafe-eval`. For example:
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/spatial-audio/project-setup/web.mdx]: done

File: shared/video-sdk/develop/spatial-audio/project-setup/web.mdx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        
+        ```html
+        <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'wasm-unsafe-eval'">
+        ```
+
+        * Versions lower than Chrome 97 and Edge 97: Add script-src in options 'unsafe-eval'.
+
+1. **Register the spatial audio effect plugin**
+
+    Call the `AgoraRTC.registerExtensions` method and pass in the created  `SpatialAudioExtension` instance.
```
[the remark this answers is not in the document]
Comment (saudsami) [reply shared/video-sdk/develop/spatial-audio/project-setup/web.mdx:43]: done

**Pull request merged**